File: cpp/labs2/lab2.2/Rectangle.cpp
#include "Rectangle.h"
#include <cmath>

Rectangle::Rectangle(double x1, double y1, double x2, double y2) : x1(x1), y1(y1), x2(x2), y2(y2) {}

double Rectangle::area() const {
    return abs((x2 - x1) * (y2 - y1));
}

double Rectangle::perimeter() const {
    return 2 * (abs(x2 - x1) + abs(y2 - y1));
}

File: csharp/lab2.2/Rectangle.cs
using System;

class Rectangle : Figure {
    private double x1, y1, x2, y2;

    public Rectangle(double x1, double y1, double x2, double y2) {
        this.x1 = x1;
        this.y1 = y1;
        this.x2 = x2;
        this.y2 = y2;
    }

    public override double Area() {
        return Math.Abs((x2 - x1) * (y2 - y1));
    }

    public override double Perimeter() {
        return 2 * (Math.Abs(x2 - x1) + Math.Abs(y2 - y1));
    }
}